### PR TITLE
Refactored test_equals_none() function

### PR DESCRIPTION
**Pull request by alexwholland**
##### SUMMARY

Removed equality statements in the `test_equals_none()` function since they are never used.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`test/units/inventory/test_host.py`

### CODE REVIEW

**Comment by webknjaz** [comment]
> since they are never used.

Well, they may produce side effects. Have you checked that? What does the original commit say?

**Comment by webknjaz** [comment]
Or maybe those lines are missing the `assert` statements, which is more likely. So I'd add the assertions instead of removing them.

File: test/units/inventory/test_host.py [hunk 1]
```diff
@@ -69,10 +69,6 @@ def test_get_groups(self):
 
     def test_equals_none(self):
         other = None
-        self.hostA == other
```
Comment (webknjaz) [inline test/units/inventory/test_host.py:72]: @alikins can you recall what these were for?

**Comment by webknjaz** [comment]
@mattclay I'm puzzled why this is happening under Python 3.9-3.11:
```python-traceback
00:53 ==================================== ERRORS ====================================
00:53 ______________ ERROR collecting test/units/inventory/test_host.py ______________
00:53 ImportError while importing test module '/root/ansible/test/units/inventory/test_host.py'.
00:53 Hint: make sure your test modules/packages have valid Python names.
00:53 Traceback:
00:53 /usr/lib/python3.10/importlib/__init__.py:126: in import_module
00:53     return _bootstrap._gcd_import(name[level:], package, level)
00:53 test/units/inventory/test_host.py:25: in <module>
00:53     from units.compat import unittest
00:53 E   ModuleNotFoundError: No module named 'units'
```

I've restarted the jobs but the change doesn't seem to be caused by this patch :thinking:.

**Comment by alexwholland** [comment]
> > since they are never used.
> 
> Well, they may produce side effects. Have you checked that? What does the original commit say?

Hi @webknjaz, as you probably saw the original commit #17827 for this test did not provide a reason for these equality statements, so I'm confused why these are here and causing the failed checks. I appreciate you looking into this issue.

**Comment by webknjaz** [comment]
> so I'm confused why these are here and causing the failed checks.

I'm actually not sure your changes are causing this. Which is why I tagged @mattclay, so we could get to the bottom of this in case it's `ansible-test` being drunk.

If the changes were to cause this, my best guess is side effects caused by the dunder methods at https://github.com/ansible/ansible/blob/045f303/lib/ansible/inventory/host.py#L43. Although, I don't see anything that could actually cause a side effect there.
Besides, it fails way earlier any tests are being invoked. pytest fails on the test collection stage, which is when it just does the discovery of what's available on disk to compile the list of tests.

**Comment by webknjaz** [comment]
@mattclay noticed that missing `test/units/inventory/__init__.py` is causing this

**Comment by webknjaz** [comment]
This will fix it https://github.com/ansible/ansible/pull/78592

**Comment by webknjaz** [comment]
@alexwholland it's green now, but I'd still go for adding `assert`s instead of removing the lines.

**Review by mattclay — APPROVED** [review]

File: test/units/inventory/test_host.py [hunk 2]
```diff
@@ -69,10 +69,10 @@ def test_get_groups(self):
...
-        other == self.hostA
-        self.hostA != other
-        other != self.hostA
+        assert self.hostA == other
+        assert other == self.hostA
+        assert self.hostA != other
+        assert other != self.hostA
         self.assertNotEqual(self.hostA, other)
```
Comment (webknjaz) [inline test/units/inventory/test_host.py]: This is covered by `assert self.hostA != other`
```suggestion
```

File: test/units/inventory/test_host.py [hunk 3]
```diff
@@ -69,10 +69,10 @@ def test_get_groups(self):
 
     def test_equals_none(self):
         other = None
-        self.hostA == other
-        other == self.hostA
-        self.hostA != other
-        other != self.hostA
+        assert self.hostA == other
+        assert other == self.hostA
```
Comment (webknjaz) [inline test/units/inventory/test_host.py]: I think this was supposed to be
```suggestion
        assert not (self.hostA == other)
        assert not (other == self.hostA)
```

**Comment by mattclay** [comment]
@alexwholland Thanks for submitting this PR. It looks like we forgot about this after the most recent changes were made. These fixes, along with other similar fixes, were merged as part of: https://github.com/ansible/ansible/pull/79940

**Pull request closed without merging**